src/main/java/com/softserve/service/impl/LessonServiceImpl.java

```diff
@@ -21,7 +21,7 @@
 import java.util.List;
 
 @Slf4j
-@Transactional
+@Transactional(readOnly = true)
```

Comment:
you can remove @Transactional from class and use it only for methods, it will be more clear. but this is just advice